mParticle-Swrve/MPKitSwrve.m

```diff
@@ -0,0 +1,383 @@
+                                                              userInfo:userInfo];
+        });
+    });
+    
+    return [self execStatus:MPKitReturnCodeSuccess];
+}
+
+/*
+    Implement this method if your SDK should be notified when the user logs in
+*/
+//- (MPKitExecStatus *)onLoginComplete:(FilteredMParticleUser *)user request:(FilteredMPIdentityApiRequest *)request {
+//     /*  Your code goes here.
+//         If the execution is not successful, please use a code other than MPKitReturnCodeSuccess for the execution status.
+//         Please see MPKitExecStatus.h for all exec status codes
+//      */
+//
+//     return [self execStatus:MPKitReturnCodeSuccess];
+//}
+
+/*
+    Implement this method if your SDK should be notified when the user logs out
+*/
+//- (MPKitExecStatus *)onLogoutComplete:(FilteredMParticleUser *)user request:(FilteredMPIdentityApiRequest *)request {
+//     /*  Your code goes here.
+//         If the execution is not successful, please use a code other than MPKitReturnCodeSuccess for the execution status.
+//         Please see MPKitExecStatus.h for all exec status codes
+//      */
+//
+//     return [self execStatus:MPKitReturnCodeSuccess];
+//}
+
+/*
+    Implement this method if your SDK should be notified when user identities change
+*/
+//- (MPKitExecStatus *)onModifyComplete:(FilteredMParticleUser *)user request:(FilteredMPIdentityApiRequest *)request {
+//     /*  Your code goes here.
+//         If the execution is not successful, please use a code other than MPKitReturnCodeSuccess for the execution status.
+//         Please see MPKitExecStatus.h for all exec status codes
+//      */
+//
+//     return [self execStatus:MPKitReturnCodeSuccess];
+//}
+
+#pragma mark e-Commerce
+/*
+    Implement this method if your SDK supports commerce events.
+    If your SDK does support commerce event, but does not support all commerce event actions available in the mParticle SDK,
+    expand the received commerce event into regular events and log them accordingly (see sample code below)
+    Please see MPCommerceEvent.h > MPCommerceEventAction for complete list
+*/
+ - (MPKitExecStatus *)logCommerceEvent:(MPCommerceEvent *)commerceEvent {
+     MPKitExecStatus *execStatus = [[MPKitExecStatus alloc] initWithSDKCode:[[self class] kitCode] returnCode:MPKitReturnCodeSuccess forwardCount:0];
+     NSString* currency = commerceEvent.currency ? : @"USD";
+
+     // In this example, this SDK only supports the 'Purchase' commerce event action
+     if (commerceEvent.action == MPCommerceEventActionPurchase) {
+             /* Your code goes here. */
+         
+         for (MPProduct *product in commerceEvent.products) {
+             SwrveIAPRewards* rewards = [[SwrveIAPRewards alloc] init];
+             [[SwrveSDK sharedInstance] unvalidatedIap:rewards localCost:[product.price doubleValue] localCurrency:currency productId:product.sku productIdQuantity:[product.quantity integerValue]];
```

Comment:
are you intentionally leaving `rewards` an empty object here? or should products and currency be added to it?

Reply:
Yeah, intentionally leaving it empty, since I'm treating the sku as the purchased item. I don't think we can really do much more here, since it's hard to assume what will be sent by the customer app.

Reply:
that sounds good to me

Reply:
oh at the last point there, it should be `[product.quantity intValue]` not `[product.quantity integerValue]`. that'll fire a warning